### PR TITLE
Update and expand developer/contributor documentation

### PR DESCRIPTION
**Pull request by brainwane**
Today I did a pass through the installation, README, design, and contribution guides. I improved a bunch of formatting and linking, and:

* added a general architecture overview to `DESIGN.md`
* made the `README.md` sections linkable and made the table of contents into hyperlinks
* made the headers in `INSTALL.md` more consistent
* updated CPU and OS requirements in `INSTALL.md` and clarified the JNDI stuff
* updated last-reviewed dates and expectations of reliability in various documents
* added a thank-you section to `CONTRIBUTING.md`

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -46,20 +46,24 @@ time, and will evolve as we understand it more.
...
 
 ### Software
 
-- **Operating System**: we recommend the stable Debian 8 (jessie). If
-  that's not feasible for your environment, any of the supported
-  WildFly 10.1 operating systems should work, but our ability to help
-  troubleshoot issues that come up may be limited.  Once we test this
-  on a few more platforms, we will expand the list of compatible
-  operating systems to include onther Linux distributions.
+- **Operating System**: we recommend the stable Debian 8 (jessie), and
```
Comment (jasonaowen) [inline INSTALL.md]: Oh, jessie is oldstable now, isn't it? We'll have to update this at some point. That probably shouldn't be in this PR, though.

File: README.md [hunk 2]
```diff
@@ -5,11 +5,11 @@ Current build status:  ![Build status](https://travis-ci.org/OpenTechStrategies/
...
 
-1.  Introduction and Current Status
-2.  Background
-3.  Features and Functionality
-4.  Project Resources and Organization of This Repository
-5.  Participating in the PSM Project
+1. **[Introduction and Current Status](#section-1-introduction-and-current-status)**
+2. **[Background](#section-2-background)**
+3. **[Features and Functionality](#section-3-features-and-functionality)**
+4. **[Project Resources and Organization of This Repository](#section-4-project-resources-and-organization-of-this-repository)**
```
Comment (jasonaowen) [inline README.md]: Bullet 4 seems not to be a valid link; viewing the rendered page tells me it should be `#section-4-project-resources-and-repository-organization`.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -22,10 +22,13 @@ issues at this stage in even simple patches.
 
 Generally, the more controversial, complex or large a change, the more
 opportunity people should have to comment on it.  That means it should
-garner more comments/approvals or it means it should sit longer before
-being merged.  We don't have hard rules about such things, and
-documentation changes usually don't need to sit as long as functional
-changes, but figure a business day or two for an average patch.
+garner more comments/approvals, or it means it should sit longer
+before being merged. You can talk with us about a change you'd like to
+make before or while you work on it; see [the participation section of
```
Comment (jasonaowen) [inline CONTRIBUTING.md]: :100: yespls!

File: INSTALL.md (hunk 1 above)
Comment (brainwane) [inline INSTALL.md]: Thanks! Will file an issue about us moving to stretch, but will also fix this line to say that 8 is old stable.

File: INSTALL.md [hunk 4]
```diff
@@ -339,7 +339,7 @@ repository, update its location in your local properties:
       ```
 
 1. To check that the app is running, navigate to
-   http://localhost:8080/cms/login.  You should see a login screen.
+   http://localhost:8080/cms/login .  You should see a login screen.
```
Comment (brainwane) [inline INSTALL.md]: Good idea. Done.

File: README.md (hunk 2 above)
Comment (brainwane) [inline README.md]: Argh, thanks. Fixed.

File: CONTRIBUTING.md (hunk 3 above)
Comment (brainwane) [inline CONTRIBUTING.md]: :)

**Comment by brainwane** [comment]
@jasonaowen I've responded to your suggestions; thanks.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -14,15 +15,20 @@ This repository has a home on
...
+the README](README.md#section-5-participating-in-the-psm-project). We
+don't have hard rules about such things, and documentation changes
+usually don't need to sit as long as functional changes, but figure a
+business day or two for an average patch to get discussed.
 
 As to when to merge, that's a judgment call.  Usually Cecilia, James
 or Karl pulls the trigger, but that isn't to say others shouldn't feel
```
Comment (brainwane) [inline CONTRIBUTING.md]: I shall do this.

File: INSTALL.md [hunk 6]
```diff
@@ -1,13 +1,14 @@
        Installation instructions for the Provider Screening Module
        ===========================================================
 
-***NOTE: 2017-04-28: These instructions are still very incomplete and
-   are a work in progress.  We welcome suggestions on improving
-   them.***
+***NOTE: 2017-07-03: These instructions are fairly reliable but
+   incomplete and a work in progress.  We welcome suggestions on
```
Comment (brainwane) [inline INSTALL.md]: I think they are -- we still don't have as much certainty on the OS, storage and CPU requirements as we'd like.

File: INSTALL.md [hunk 7]
```diff
@@ -17,7 +18,7 @@ state's MMIS environment.  Our plan is to retarget the PSM to WildFly
 (formerly JBoss), though still keeping all of the functionality
```
Comment (brainwane) [inline INSTALL.md]: Fixing.

File: INSTALL.md (hunk 6 above)
Comment (jasonaowen) [inline INSTALL.md]: From a discussion in IRC:

Some of our requirements are vague, and we intentionally do not have a production deployment guide yet - that will need to be developed by talking with state operations engineers. However, our instructions are complete enough for a development or test deployment, and we no longer need a big scary warning.

File: INSTALL.md [hunk 8]
```diff
@@ -46,20 +46,24 @@ time, and will evolve as we understand it more.
...
-  troubleshoot issues that come up may be limited.  Once we test this
-  on a few more platforms, we will expand the list of compatible
-  operating systems to include onther Linux distributions.
+- **Operating System**: we recommend the old stable Debian 8 (jessie), and
+  a developer has successfully installed the PSM on Red Hat 7.3
+  Enterprise Linux. If that's not feasible for your environment, any
+  of the supported WildFly 10.1 operating systems should work, but our
+  ability to help troubleshoot issues that come up may be limited.
+  Once we test this on a few more platforms, we will expand the list
+  of compatible operating systems to include onther Linux
```
Comment (jasonaowen) [inline INSTALL.md]: Oh, I just noticed a typo that's been there a while: `onther`. That was probably me; sorry!

Comment (brainwane) [inline INSTALL.md]: Fixed, thanks!

**Pull request merged**